src/compiler/commandLineParser.ts

```diff
@@ -961,12 +961,15 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
     {
         name: "moduleResolution",
         type: new Map(getEntries({
-            node: ModuleResolutionKind.NodeJs,
+            // N.B. The first entry specifies the value shown in `tsc --init`
+            node10: ModuleResolutionKind.Node10,
+            node: ModuleResolutionKind.Node10,
             classic: ModuleResolutionKind.Classic,
             node16: ModuleResolutionKind.Node16,
             nodenext: ModuleResolutionKind.NodeNext,
             bundler: ModuleResolutionKind.Bundler,
         })),
+        deprecatedKeys: new Set(["node"]),

```

Comment:
Right now, this only filters the config parsing diagnostic that says what the valid options are.